src/test/java/io/jenkins/plugins/analysis/core/model/ExcludeCategoryTest.java

```diff
@@ -0,0 +1,136 @@
+    }
+
+    /**
+     * Verifies that an empty filter is handled correctly.
+     */
+    @Test
+    void shouldExcludeNoIssueWhenPatternIsEmpty() {
+        final Predicate<? super Issue> filter = buildIssueFilterBuilderWithGivenPatternStructure(
+                new IssueFilterBuilder(), getIssuesFilterInstance(),
+                EMPTY_PATTERN);
+
+        applyFilterAndCheckResult(filter, getIssues(), true, ISSUE1, ISSUE2, ISSUE3);
+    }
+
+    /**
+     * Verifies that a filter covering all possibilities is handled correctly.
+     */
+    @Test
+    void shouldPassNoneWhenAGeneralFilterIsAdded() {
+        final Predicate<? super Issue> filter = buildIssueFilterBuilderWithGivenPatternStructure(
+                new IssueFilterBuilder(), getIssuesFilterInstance(),
+                "[a-zA-Z1-3]*");
+
+        applyFilterAndCheckResult(filter, getIssues(), true);
+    }
+
+    /**
+     * Verifies that a filter covering no possibilities is handled correctly.
+     */
+    @Test
+    void shouldPassAllWhenUselessFilterIsAdded() {
+        final Predicate<? super Issue> conditionWithMultiplePattern = buildIssueFilterBuilderWithGivenPatternStructure(
+                new IssueFilterBuilder(), getIssuesFilterInstance(), "[a-zA-Z99]*", "[a-zA-Z98]*", "[a-zA-Z97]*");
+
+        applyFilterAndCheckResult(conditionWithMultiplePattern, getIssues(), true, ISSUE1, ISSUE2, ISSUE3);
+    }
+
+    /**
+     * Verifies that a NullPointerException is thrown in case the pattern is null.
+     */
+    @Test
+    void shouldThrowNullPointerExceptionForAPatternWhichIsNull() {
+        final Predicate<? super Issue> filter = buildIssueFilterBuilderWithGivenPatternStructure(
+                new IssueFilterBuilder(), getIssuesFilterInstance(),
+                (String) null);
+
+        assertThatNullPointerException().isThrownBy(() -> applyFilterAndCheckResult(filter, getIssues(), false))
+                .withNoCause();
+    }
+
+    /**
+     * Verifies that a NullPointerException is thrown in case the issues filter is null.
+     */
+    @Test
+    void shouldThrowNullPointerExceptionForAnIssueFilterBuilderWhichIsNull() {
+        assertThatNullPointerException().isThrownBy(
+                () -> buildIssueFilterBuilderWithGivenPatternStructure(new IssueFilterBuilder(), null,
+                        FILTER_NAME_TO_CHECK))
+                .withNoCause();
+    }
+
+    /**
+     * Verifies that a NullPointerException is thrown in case the issues filter and the pattern is null.
+     */
+    @Test
+    void shouldThrowNullPointerExceptionForAnIssueFilterBuilderAndAPatternWhichAreNull() {
+        assertThatNullPointerException().isThrownBy(
+                () -> buildIssueFilterBuilderWithGivenPatternStructure(new IssueFilterBuilder(), null, (String) null))
+                .withNoCause();
+    }
+
+    /**
+     * Verifies that the class implementing serializable is serializable.
+     */
+    @Test
+    void shouldBeSerializable() throws IOException {
```

Comment:
Da gibt es schon eine Baseclass die man verwenden könnte: SerializableTest.
Alternativ den ganzen Test in die Oberklasse auslagern und die getIssuesFilterInstance als protected abstract im Parent.